### ISSUE TITLE
keyboard keys do not work when we do $('id').select2('open')

### ISSUE THREAD

**Issue by sidharthc** [select2/select2#3680]
i open the dropdown by this code:

```
$("#selectId").select2("open");
$("#selectId").select2({
  minimumResultsForSearch: Infinity,
  dropdownAutoWidth: 'true'
});
```

keyboard keys do not work to navigate through the options.

 But when I remove minimumResultsForSearch: Infinity, I am able to use the keys.
I am using version select 4.0.0

**Comment by ggely** [comment]
I have the same problem, but if you see the example "Hiding the search box" in the demo page (https://select2.github.io/examples.html), when you set minimumResultsForSearch to Infinity, the search functionality (enter some letter to find an entry) doesn't work. In this case i think the search module is badly initialized and doesn't catch the event.

**Comment by kevin-brown** [comment]
The problem appears to be that the focus is not always being set to the Select2 container, so it isn't able to catch the key events.

http://jsbin.com/lasurazeca/1/edit?html,js,output

Note that this isn't an issue when the search box is there, because the search box is automatically focused (after a 1ms delay) so it will always steal the focus back.

http://jsbin.com/zonehadofe/1/edit?html,js,output

**Comment by alexweissman** [comment]
This issue is discussed more thoroughly in #4713 - please continue the conversation there.
